TESTING.md

```diff
@@ -0,0 +1,69 @@
+## Testing
+
+`canvas-kit` has 3 levels of testing: unit, functional and visual. Each area has special use-cases even though tests at various levelsmay overlap. Here's a simplified explaination of what each level is meant to test:
+
+- *unit*: This level is meant to test the expected outputs of developer APIs (ex: When a developer passes in X prop, then expect a specific output
+- *functional*: This level is browser-based testing and is where UI behavior and accessibility requirements are specified
+- *visual*: This level is where all styling is tested. Some props have only visual changes and it is difficult to reason about a in any other level
+
+Tests are supposed to be a source of feedback and confidence. To acheive that, a test should be clear about intent and failures. Developers tend to ignore passing tests and focus only on failing tests. If the intent of the test cannot be determined, it is difficult to fix the test. If the failure isn't immediately obvious, too much time is spent fixing the test. Canvas-kit adopted a BDD (behavior-driven development) approach of testing. The most common way to think is about behavior is "given/when/then". These behaviors are a seriees of specifications and not necessarily "tests". Testing in this repo is not just testing the correctness of our applications, but running a series of specifications. Specifications tend to be more granular to give context of what code is expected to do. A "test" simply has to pass. A specification requires meaning.
```

Comment:
```suggestion
Tests are supposed to be a source of feedback and confidence. To achieve that, a test should be clear about intent and failures. Developers tend to ignore passing tests and focus only on failing tests. If the intent of the test cannot be determined, it is difficult to fix the test. If the failure isn't immediately obvious, too much time is spent fixing the test. Canvas Kit adopts a BDD (behavior-driven development) approach of testing. The most common way to think is about behavior is "given/when/then". These behaviors are a series of specifications and not necessarily "tests". Testing in this repo is not just testing the correctness of our components, but running a series of specifications. Specifications tend to be more granular to give context of what code is expected to do. A "test" simply has to pass. A specification requires meaning.
```

Reply:
Yikes, diffs don't work here. I just made a couple of typo changes ("acheive" -> "achieve" and "seriees" -> "series"), changed "Canvas-kit" to "Canvas Kit" (which is our preferred proper noun for the repo), and changed the tense of "adopted" to "adopts" since it sounds more authoritative.

Reply:
And... Amen, this is a great paragraph. ✊